### PR TITLE
task: refactor and test temp module

### LINKED ISSUE #8: task: refactor and test temp module
### Description:

Refactor the temp module of the mainboard and add the missing unit tests.

### Implementation steps:

1. Change all the identifiers name following the embedded conventions
2. Implement unit tests
3. Fix small issues if any

### Dependencies:

None

### Additional notes:

_No response_

### PR DESCRIPTION
**Pull request by ale-giustina**
# Purpose
Refactoring of temp module

# Overview
Removed typedef on structs and enums, added doxygen comments and added unit tests.

# Guidance
The main files that have been affected are the header files in their dedicated folder, their implementation and the tests. Other files may have been modified marginally due to variable renaming.

Closes #8

### CODE REVIEW

File: Core/Inc/bms/temp/temp.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+#include "primary_network.h"
+#include "bms_network.h"
+
+/*! \brief Minimum and maximum allowed cell temperature in celsius */
+#define TEMP_MIN_C (-10.f)
+#define TEMP_MAX_C (60.f)
+
+/*! \brief Number of temperatures sent in a single temp can message */
+#define TEMP_NUM_TEMP_CAN_MESSAGE (4U)
```
Comment (Tonidotpy) [inline Core/Inc/bms/temp/temp.h]: Rename to `TEMP_TEMPERATURE_PER_MESSAGE_COUNT` for clarity.

File: Core/Inc/bms/temp/temp-api.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+primary_hv_cells_temp_stats_converted_t *temp_api_get_cells_temperature_stats_canlib_payload(size_t *byte_size);
+
+#else
+
+#define temp_api_init() (TEMP_OK)
+#define temp_api_get_values() (NULL)
+#define temp_api_get_min() (NULL)
+#define temp_api_get_max() (NULL)
+#define temp_api_get_avg() (NULL)
+#define temp_api_cells_temperature_handle(payload) MAINBOARD_NOPE()
```
Comment (Tonidotpy) [inline Core/Inc/bms/temp/temp-api.h]:
```suggestion
#define temp_api_cells_temperature_handle(payload) EAGLETRT_API_NOP()
```

File: Core/Src/bms/temp-api.c [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+EAGLETRT_STATIC char *temp_return_code_name[] = {
+    [TEMP_OK] = "ok",
+    [TEMP_NULL_POINTER] = "null pointer",
+    [TEMP_OUT_OF_BOUNDS] = "out of bounds"
+};
+
+EAGLETRT_STATIC char *temp_return_code_description[] = {
+    [TEMP_OK] = "executed successfully",
+    [TEMP_NULL_POINTER] = "attempt to dereference a null pointer",
+    [TEMP_OUT_OF_BOUNDS] = "attempt to access an invalid memory region"
```
Comment (Tonidotpy) [inline Core/Src/bms/temp-api.c]: Missing `RC` prefix.

File: test/test_temp/test-temp.c [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+}
+
+void test_temp_api_get_cells_temperature_canlib_payload_offset_advances(void) {
+    temp_handler.cellboard_id = 0U;
+    temp_handler.offset = 0U;
+
+    (void)temp_api_get_cells_temperature_canlib_payload(NULL);
+
+    TEST_ASSERT_EQUAL_MESSAGE(TEMP_NUM_TEMP_CAN_MESSAGE, temp_handler.offset, "temp_api_get_cells_temperature_canlib_payload() did not advance offset");
+}
```
Comment (Tonidotpy) [inline test/test_temp/test-temp.c]: Testing same behavior, can be merged.

File: test/test_temp/test-temp.c [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+    }
+    temp_handler.temperatures[0U][0U] = 5.0F;
+    temp_handler.temperatures[1U][0U] = 60.0F;
+
+    primary_hv_cells_temp_stats_converted_t *payload = temp_api_get_cells_temperature_stats_canlib_payload(NULL);
+
+    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(0.01F, 5.0F, payload->min, "temp_api_get_cells_temperature_stats_canlib_payload() wrong min");
+    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(0.01F, 60.0F, payload->max, "temp_api_get_cells_temperature_stats_canlib_payload() wrong max");
+    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(0.1F, temp_api_get_avg(), payload->avg, "temp_api_get_cells_temperature_stats_canlib_payload() wrong avg");
+}
```
Comment (Tonidotpy) [inline test/test_temp/test-temp.c]: Can be merged.

File: test/test_temp/test-temp.c
Comment (Tonidotpy) [inline test/test_temp/test-temp.c:1]: Missing test for `get_canlib_payload` functions with `NULL` byte size.

**Review by Tonidotpy — CHANGES_REQUESTED** [review]

**Review by Tonidotpy — APPROVED** [review]
